### ISSUE TITLE
SessionProtect = named fails when using -fork

### ISSUE THREAD

**Issue by magnumripper** [magnumripper/JohnTheRipper#1764]
If `SessionProtect = named`, and using -fork, and there are existing files eg. `john.2.rec`, John will erroneously terminate (as if `SessionProtect = always`) because it doesn't recognize the "numbered" default file name.

**Comment by frank-dittrich** [comment]
The correct fix is to only do the check for the main process, not for forked processes.

You can't --restore=test.2, if test.rec doesn't exist.

**Comment by magnumripper** [comment]
Thanks, that's a simple solution. I'll try it out.

**Comment by magnumripper** [comment]
That did not quite work. The main process aborted but the child resumed (needing a manual kill). I think I really need to compare with a numbered default.

**Comment by magnumripper** [comment]
Also found and fixed another bug: @jfoug had hard-coded the file name, but it's actually made up from macros in params.h and differs for JOHN_SYSTEMWIDE builds.

**Comment by frank-dittrich** [comment]
I still think that the right fix is to implement that check just for the main process.
May be it must be moved before the fork.

**Comment by magnumripper** [comment]
Yes, if it's only for the main process it HAS to be prior to fork.

The resuming stuff is incredibly hard to follow, I've been fighting it for years and still don't quite get it. Sometimes I think I do get it, then I'm immediately proven wrong. So I'm not touching this now until we see any problems :smile:
